cpp/src/link_analysis/pagerank_1D.cu

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Author: Alex Fender afender@nvidia.com
+
+#include <graph.hpp>
+#include "pagerank_1D.cuh"
+#include "utilities/graph_utils.cuh"
+
+namespace cugraph {
+namespace opg {
+
+#define CUDA_MAX_KERNEL_THREADS 256
+#define CUDA_MAX_BLOCKS 65535
+
+template <typename VT, typename ET, typename WT>
+__global__ void __launch_bounds__(CUDA_MAX_KERNEL_THREADS)
+  transition_kernel(const size_t e, const VT *ind, const VT *degree, WT *val)
+{
+  for (auto i = threadIdx.x + blockIdx.x * blockDim.x; i < e; i += gridDim.x * blockDim.x)
+    val[i] = 1.0 / degree[ind[i]];
+}
```

Comment:
What happens if degree[ind[i]] == 0? Doesn't matter as this will not be used?

Reply:
You should probably use bookmark flag set in flag_leaf to avoid calculation when degree[ind] is 0.

Reply:
Isn't that require additional memory for bookmark? We'd better avoid that. Here the performance will be likely to be limited by getting degree[ind[i]] (random access) and having a predicated instruction (to check degree is 0) will have only very minor impact on performance (or can be faster by avoiding additional memory access to get bookmark). Better avoid unnecessarily increasing the memory footprint.

Reply:
> What happens if degree[ind[i]] == 0? 

This can't happen mathematically.

Reply:
Degree contains `IN` degree. So all `degree[ind[i]]` where incremented by definition.

Reply:
Could you add a comment here? This is non-trivial to figure out, so comment will be helpful if others need to maintain this code.

Reply:
Added a comment